tools/disasm/disasm.py

```diff
@@ -2,126 +2,260 @@
-constants = {}         # lui/ori pairs : {addr of %hi : full constant} AND {addr of %lo : full constant} , twice the space complexity but much less time complexity
-dwords = set()         # doublewords
-multiply_referenced_rodata = set() # rodata with more than 1 reference outside of the same function cannot be migrated
-
-files = set()          # vram start of file
-
-vrom_variables = list() # (name,addr)
+jtbls = set()  # vram of jump tables
+jtbl_labels = (
+    set()
+)  # vram of branch labels via jtbls, higher output priority than regular branch labels
+floats = set()  # vram of floats
+doubles = set()  # vram of doubles
+prospective_strings = set()  # vram of possible strings
+strings = set()  # vram of confirmed strings
+symbols = (
+    {}
+)  # lui/addiu pairs : {addr of %hi : full symbol} AND {addr of %lo : full symbol} , twice the space complexity but much less time complexity
+constants = (
+    {}
+)  # lui/ori pairs : {addr of %hi : full constant} AND {addr of %lo : full constant} , twice the space complexity but much less time complexity
+dwords = set()  # doublewords
+multiply_referenced_rodata = (
+    set()
+)  # rodata with more than 1 reference outside of the same function cannot be migrated
+
+files = set()  # vram start of file
+
+vrom_variables = list()  # (name,addr)
 
 functions_ast = None
 variables_ast = None
 
-vars_cache = {} # (address: variable + addend)
+vars_cache = {}  # (address: variable + addend)
+
+files_text = {}
+
 
 def proper_name(symbol, in_data=False, is_symbol=True):
     # hacks
-    if symbol == 0x809C46F0: # ovl_En_Encount4 fake symbol at the very end of the data section
+    if (
+        symbol == 0x809C46F0
+    ):  # ovl_En_Encount4 fake symbol at the very end of the data section
         return variables_ast[0x809C46DC][0] + " + 0x14"
-    elif symbol == 0x801EF66D: # z_message_nes constant-folding stray fairy array
+    elif symbol == 0x801EF66D:  # z_message_nes constant-folding stray fairy array
         return variables_ast[0x801EF670][0] + f" - 0x{0x801EF670 - 0x801EF66D:X}"
-    elif symbol == 0x80A09740: # boss_07 symbol with large addend folded into %lo
+    elif symbol == 0x80A09740:  # boss_07 symbol with large addend folded into %lo
         return variables_ast[0x80A09A60][0] + f" - 0x{0x80A09A60 - 0x80A09740:X}"
-    elif symbol == 0x80B80248: # bg_ikana_mirror symbol with large addend folded into %lo
+    elif (
+        symbol == 0x80B80248
+    ):  # bg_ikana_mirror symbol with large addend folded into %lo
         return variables_ast[0x80B801A8][0] + f" + 0x{0x80B80248 - 0x80B801A8:X}"
-    elif symbol == 0x8084D2FC: # player symbol with very large addend folded into %lo, since we don't know the real symbol just use the first data symbol for now
+    elif (
+        symbol == 0x8084D2FC
+    ):  # player symbol with very large addend folded into %lo, since we don't know the real symbol just use the first data symbol for now
```

Comment:
```suggestion
    # player symbol with very large addend folded into %lo, since we don't know the real symbol just use the first data symbol for now
    elif symbol == 0x8084D2FC:
```